### PR TITLE
Replace eastwood with clj-kondo

### PR DESCRIPTION
**Pull request by ezmiller**
## Goal

Replace the linter eastwood with a more popular and polished linter: `clj-kondo`: https://github.com/clj-kondo/clj-kondo

## Also in this PR

Miscellaneous changes to get the linters to pass.

### CODE REVIEW

File: .clj-kondo/config.edn [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{:linters
+ {:unresolved-var {:level :off} ;; set b/c seems to be triggered by fns exported with tech.datatype.export-symbols/export-symbols fn.
```
Comment (ezmiller) [inline .clj-kondo/config.edn:2]: Wish we didn't have to turn this off, but until we better understand `export-symbols`, and if there are other ways of defining the api, it seems to be caused by that.

File: project.clj [hunk 2]
```diff
@@ -8,10 +8,15 @@
...
+                        [aerial.hanami "0.12.4"]
+                        [clj-kondo "2021.03.03"]
+                        [midje/midje "1.9.10"
+                          :exclusions [org.clojure/clojure]]]
+         :plugins [[lein-cljfmt "0.7.0"]
+                   [lein-midje "3.2.1"]]
+         :aliases {"clj-kondo" ["run" "-m" "clj-kondo.main"]
+                   "lint" ["do"
+                           ["cljfmt" "check"]
+                           ["run" "-m" "clj-kondo.main" "--lint" "src:test"]]}}})
```
Comment (ezmiller) [inline project.clj:22]: Note: change here is to lint **both** `./src` and `./test`.

File: test/tablecloth/time/api/slice_test.clj [hunk 3]
```diff
@@ -1,9 +1,10 @@
 (ns tablecloth.time.api.slice-test
-  (:require [tablecloth.api :refer [dataset columns column-names]]
+  (:require [tablecloth.api :refer [dataset]]
             [tablecloth.time.index :refer [index-by]]
             [tablecloth.time.api :refer [slice]]
             [tech.v3.datatype.datetime :refer [long-temporal-field plus-temporal-amount]]
-            [clojure.test :refer [deftest is are]]))
+            [clojure.test :refer [deftest is are]]
+            [time-literals.data-readers]))
```
Comment (ezmiller) [inline test/tablecloth/time/api/slice_test.clj:7]: Need to explicitly load these time-literals b/c tick is not in the namespace. See issue #21.

**Review by rsiddharthan — APPROVED** [review]
Looks fine. Need to adopt linter config locally to be in sync

**Comment by ezmiller** [comment]
> Looks fine. Need to adopt linter config locally to be in sync

@rsiddharthan the config should come down when you pull down the repository. What you may want to do is install clj-kondo globally on your computer and so that you can run it in your editor. Here's a link to the a guide on that setup. It's not too bad: https://github.com/clj-kondo/clj-kondo/blob/master/doc/editor-integration.md

**Pull request merged**